src/libsystemd/sd-netlink/netlink-message.c

```diff
@@ -15,8 +15,7 @@
 #include "socket-util.h"
 #include "strv.h"
 
-#define GET_CONTAINER(m, i) ((i) < (m)->n_containers ? (struct rtattr*)((uint8_t*)(m)->hdr + (m)->containers[i].offset) : NULL)
-#define PUSH_CONTAINER(m, new) (m)->container_offsets[(m)->n_containers++] = (uint8_t*)(new) - (uint8_t*)(m)->hdr;
+#define GET_CONTAINER(m, i) ((struct rtattr*)((uint8_t*)(m)->hdr + (m)->containers[i].offset))
```

Comment:
why this change? the old macro looks more robust? can you elaborate?

Reply:
- `PUSH_CONTAINER()` is unused, and maybe not in sync with code, too dangerous to keep it.
 - Nobody check the return value of `GET_CONTAINER()` (no check for NULL), and this macro is always called with a previous check on `i` with `n_containers`.

Reply:
ok